### ISSUE TITLE
How do I get turnbinds working on linux

### ISSUE THREAD

**Issue by Hades7500** [t5mat/turnbinds#4]
I'm on Garuda Linux (Arch Based)
What do I have to do after running  `./build` ?
I found a `turnbind.exe` in the .build folder and I ran it with wine and changed the settings but it doesn't work in-game

**Comment by loocea69** [comment]
This will never work on linux because wine can't convert windows syscalls to linux api syscalls, wine can't do much with <windows.h>.

**Comment by t5mat** [comment]
I'm not sure what makes this not work under Wine; I do not have a Linux environment for testing this atm. I might look into this in the future though.

**Comment by CallMeVerity** [comment]
> I'm on Garuda Linux (Arch Based) What do I have to do after running `./build` ? I found a `turnbind.exe` in the .build folder and I ran it with wine and changed the settings but it doesn't work in-game

Extremely late to the party but I added Linux support in my fork: https://github.com/CallMeVerity/turnbinds

Build with cmake -B build && cmake --build build, then run ./build/turnbinds.
For inputs to work you need to run sudo usermod -aG input $USER and log out/back in.

Note: mouse movement requires X11 (or XWayland), so it works in games running under XWayland on Wayland compositors like Hyprland.

**Comment by loocea69** [comment]
I've made turnbinds that run regardless if you're on X11 or Wayland, though requires sudo to listen to key inputs and some manual configuration to address the right devices.

**Comment by CallMeVerity** [comment]
> I've made turnbinds that run regardless if you're on X11 or Wayland, though requires sudo to listen to key inputs and some manual configuration to address the right devices.

Are you using uinput? In testing I could get mouse movement to work but the side buttons did not register properly and so I went with this approach.

**Comment by loocea69** [comment]
Yes I'm using uinput. Side buttons aren't working by default because they are not in the source code, though you could add this manually ^^
